packager/packager.cc

```diff
@@ -1542,6 +1556,83 @@ unique_ptr<PackageInfoImpl> createAndPopulatePackageInfo(core::GlobalState &gs,
     return info;
 }
 
+// Metadata for Tarjan's algorithm
+struct ComputeSCCsMetadata {
+    int nextIndex = 1;
+    int nextSCCId = 0;
+    // A given package's index in the DFS traversal; ie. when it was first visited
+    UnorderedMap<core::packages::MangledName, int> index;
+    // The lowest index reachable from a given package (in the same SCC) by following any number of tree edges
+    // and at most one back/cross edge
+    UnorderedMap<core::packages::MangledName, int> lowLink;
+    UnorderedMap<core::packages::MangledName, bool> onStack;
+    std::vector<core::packages::MangledName> stack;
+};
+
+// DFS traversal for Tarjan's algorithm starting from pkgName, along with keeping track of some metadata needed for
+// detecting SCCs.
+void strongConnect(core::GlobalState &gs, ComputeSCCsMetadata &metadata, core::packages::MangledName pkgName) {
+    if (!gs.packageDB().getPackageInfo(pkgName).exists()) {
+        return;
+    }
+    auto &pkgInfo = PackageInfoImpl::from(gs.packageDB().getPackageInfo(pkgName));
+    metadata.index[pkgName] = metadata.nextIndex;
+    metadata.lowLink[pkgName] = metadata.nextIndex;
+    metadata.nextIndex++;
+    metadata.stack.push_back(pkgName);
+    metadata.onStack[pkgName] = true;
+    for (auto &i : pkgInfo.importedPackageNames) {
+        if (i.type == ImportType::Test) {
+            continue;
+        }
```

Comment:
Do we mention anything about this SCC / strict deps analysis completely ignoring test imports in the docs?

Reply:
Added a note to the error reference that `test_import`s are skipped.